ffi/blitbuffer.lua

```diff
@@ -1826,6 +1826,216 @@ function BB_mt.__index:writePNGFromBGR(filename)
     C.free(cdata)
 end
 
+function BB_mt.__index:writeBMP(filename)
+    local function write_uint32(target_ptr, target_pos, data)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = band(data, 255)
+        target_pos = target_pos + 1
+        data = math.floor(data / 256)
+        target_ptr[target_pos] = data
+        return target_pos + 1
+    end
+
+    local output_channels = 3
+    local w, h = self:getWidth(), self:getHeight()
+    local stride = w * output_channels
+
+    -- we do this before all buffer alocations
+    local of, err = io.open(filename, "wb")
+    if err ~= nil then
+        return err
+    end
+
+    local bbdump
+    local data
+    if self:getType() == TYPE_BBRGB24 then
+        data = ffi.cast("unsigned char *", self.data)
+    else
+        bbdump = BB.new(w, h, TYPE_BBRGB24, nil)
+        bbdump:blitFrom(self)
+        data = ffi.cast("unsigned char *", bbdump.data)
+    end
+
+    local filesize = stride * h + 54
+    local padding = 0
+    -- update filesize, if stride is not a multiple of 4
+    if band(stride, 3) ~= 0 then
+        padding = 4 - band(stride, 3)
+        filesize = filesize + padding * h
+    end
+
+    local target_ptr = ffi.cast("unsigned char*", C.calloc(filesize, 1))
+    local target_pos = 0
+
+    -- bfType (2 Bytes)
+    target_ptr[target_pos] = 66 -- 'B'
+    target_pos = target_pos + 1
+    target_ptr[target_pos] = 77 -- 'M'
+    target_pos = target_pos + 1
+    -- bfSize (4 Bytes)
+    target_pos = write_uint32(target_ptr, target_pos, filesize)
+    -- bfReserved (4 Bytes)
+    target_pos = write_uint32(target_ptr, target_pos, 0)
+    -- bfOffBits (4 Byte)
+    target_pos = write_uint32(target_ptr, target_pos, 54)
+
+    -- biSize (4 Byte)
+    target_pos = write_uint32(target_ptr, target_pos, 40)
+    -- biWidth (4 Bytes)
+    target_pos = write_uint32(target_ptr, target_pos, w)
+     -- biHeight (4 Bytes)
+    target_pos = write_uint32(target_ptr, target_pos, h)
+    -- biPlanes ( 2 Bytes)
+    target_ptr[target_pos] = 1
+    target_pos = target_pos + 1
+    target_ptr[target_pos] = 0
+    target_pos = target_pos + 1
+    -- biBitCount (2 Bytes)
+    target_ptr[target_pos] = output_channels * 8
+    target_pos = target_pos + 1
+    target_ptr[target_pos] = 0
+    target_pos = target_pos + 1
+
+    for i = 1, 24 do
+        target_ptr[target_pos] = 0
+        target_pos = target_pos + 1
+    end
```

Comment:
Same idea here, I'd keep it commented with a comment that those few remaining fields belong to the header, but we don't care about 'em, so leave them set to 0.

Reply:
Without forgetting the a `target_pos =  target_pos + 24` ;p.